Scripting.md

```diff
@@ -0,0 +1,224 @@
+# Scripting
+
+This version of noggit has support for "Script Brushes", a special type of brush with scripted behavior.
+Script brushes allow us to manipulate heightmaps, textures, vertex colors, water and m2/wmo objects in a customized manner.
+Script brushes is an experimental feature, so please be careful and keep regular backups of your work when working with them.
+
+See [this video](https://youtu.be/J5-8zQ5U8gY) for an introduction to the UI.
+
+
+## Introduction to Da Scripting
+
+The scripting language we use is called "Da". It's a statically typed language that is otherwise very similar to Python, 
+and is thus indention-based but has optional support for C-style brackets too. You can use the below guide to quickly learn how to write in Da,
+
+We recommend that you use 
+[VSCodium](https://vscodium.com/#install) 
+with our 
+[minimal da extension](https://github.com/tswow/das-minimal-extension/releases/tag/v0.0.1) 
+to easily get autocompletion of function names and type specifications for the Noggit API.
+
+To create a new script brush, create a `.das` file in the `scripts` subdirectory in your noggit installation.
+We can register events by special named functions. The below code illustrates the bare minimum required for a simple left-click event with a custom title:
+
+```py
+require noggit
+
+[export]
+def title(): string
+  return "Hello World Brush"
+
+[export]
+def left_click()
+    print("Hello World!")
+```
+
+### Events
+
+All event listeners are created as global functions with the `[export]` decorator, and none of them take any arguments. 
+
+- `left_click`: Fires when the user first presses the left mouse button in the world.
+- `left_hold`: Fires continuously as the user holds the left mouse button in the world.
+- `left_release`: Fires when the user releases the left mouse button.
+- `right_click`, `right_hold`, `right_release`: Right mouse button versions of the above three functions. 
+- `select`: Fires when the user selects this script in the tool window. This event is used to register custom brush options
+- `title`: Only used to return a display name for this brush in the tool window. It must have the signature `def title(): string` if implemented.
+
+
+### Quickstart for Da syntax and types
```

Comment:
I think I should restructure the document a little differently for that (since users will still be unfamiliar with dascript in 99.99% of the cases), I think it might be better to have one tutorial and one documentation part, probably in separate documents.